modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/helper/WikiSocialActivityHelper.java

```diff
@@ -48,6 +53,46 @@ public WikiSocialActivityHelper(WikiRequestHelper wikiRequestHelper) {
 		_wikiRequestHelper = wikiRequestHelper;
 	}
 
+	public List<SocialActivity> getApprovedSocialActivities(
+			WikiPage wikiPage, int start, int end)
+		throws PortalException {
+
+		WikiPage latestWikiPage = WikiPageLocalServiceUtil.getLatestPage(
+			wikiPage.getResourcePrimKey(), WorkflowConstants.STATUS_ANY, false);
+
+		if (latestWikiPage.getPageId() == wikiPage.getPageId()) {
+			return SocialActivityLocalServiceUtil.getActivities(
+				0, WikiPage.class.getName(), wikiPage.getResourcePrimKey(),
+				start, end);
+		}
+
+		String version = String.valueOf(wikiPage.getVersion());
+
+		if (Math.floor(wikiPage.getVersion()) == wikiPage.getVersion()) {
+			version = String.valueOf((int)wikiPage.getVersion());
+		}
```

Comment:
Just a question, not a change request: Why are you taking precission errors into account in this particular case? I understand the problems with the use of doubles here, but I don't see why handling this case improves things overall (errors due to unrepresentable values may creep out anyway).

Reply:
Due to how JSONObject.toString handles type double as a whole number, the ".0" from the value is truncated before being stored in the SocialActivity.extraData column.  We may end us showing unapproved attachments, as they are stored as version "1" and not "1.0".  This only applies to whole numbers afaik, so I added this small conditional.

Reply:
Thanks!